fem/pgridfunc.cpp

```diff
@@ -683,14 +683,16 @@ void ParGridFunction::ProjectBdrCoefficient(
    gcomm.Reduce<int>(values_counter.HostReadWrite(), GroupCommunicator::Sum);
    // Accumulate the values globally.
    gcomm.Reduce<real_t>(values.HostReadWrite(), GroupCommunicator::Sum);
-   // Only the values in the master are guaranteed to be correct!
+
    for (int i = 0; i < values.Size(); i++)
    {
       if (values_counter[i])
       {
          (*this)(i) = values(i)/values_counter[i];
       }
    }
+   // Broadcast values to other processors to have a consistent GridFunction
+   gcomm.Bcast<double>((*this).HostReadWrite());
```

Comment:
There are still some places like this where `double` has to be updated to `real_t`.